src/app/messages-page/messages-inbox/messages-inbox.component.ts

```diff
@@ -51,6 +82,190 @@ export class MessagesInboxComponent implements OnInit, OnChanges {
     }
   }
 
+  showMoreButton() {
+    return !(this.newMessagesFromPage != null && this.newMessagesFromPage < this.globalVars.messagesPerFetch);
+  }
+
+  showLoadingMessage() {
+    return this.fetchingMoreMessages;
+  }
+
+  loadInitialMessages() {
+    if (!this.globalVars.loggedInUser) {
+      return;
+    }
+
+    this.backendApi.GetMessages(
+      this.appData.localNode,
+      this.globalVars.loggedInUser.PublicKeyBase58Check,
+      "",
+      this.globalVars.messagesPerFetch,
+      this.globalVars.messagesRequestsHoldersOnly,
+      this.globalVars.messagesRequestsHoldingsOnly,
+      this.globalVars.messagesRequestsFollowersOnly,
+      this.globalVars.messagesRequestsFollowedOnly,
+      this.globalVars.messagesSortAlgorithm,
+    ).subscribe(
+      (res) => {
+        if (this.globalVars.pauseMessageUpdates) {
+          // We pause message updates when a user sends a messages so that we can
+          // wait for it to be sent before updating the thread.  If we do not do this the
+          // temporary message place holder would disappear until "GetMessages()" finds it.
+        } else {
+          this.globalVars.messageResponse = res;
+
+          // Update the number of new messages so we know when to stop scrolling
+          this.newMessagesFromPage = res.OrderedContactsWithMessages.length;
+        }
+      },
+      (err) => {
+        console.error(this.backendApi.stringifyError(err));
+      }
+    );
+  }
+
+  loadMoreMessages() {
+    // If we're currently fetching messages
+    if (this.fetchingMoreMessages) {
+      return;
+    }
+
+    this.fetchingMoreMessages = true;
+    if (!this.globalVars.loggedInUser) {
+      return;
+    }
+
+    if (this.newMessagesFromPage != null && this.newMessagesFromPage == 0) {
+      return
+    }
+
+    let fetchAfterPubKey = "";
+    if (this.globalVars.messageResponse.OrderedContactsWithMessages) {
+      fetchAfterPubKey =
+        this.globalVars.messageResponse.OrderedContactsWithMessages[this.globalVars.messageResponse.OrderedContactsWithMessages.length - 1].PublicKeyBase58Check;
+    }
+
+    this.backendApi
+      .GetMessages(
+        this.appData.localNode,
+        this.globalVars.loggedInUser.PublicKeyBase58Check,
+        fetchAfterPubKey,
+        this.globalVars.messagesPerFetch,
+        this.globalVars.messagesRequestsHoldersOnly,
+        this.globalVars.messagesRequestsHoldingsOnly,
+        this.globalVars.messagesRequestsFollowersOnly,
+        this.globalVars.messagesRequestsFollowedOnly,
+        this.globalVars.messagesSortAlgorithm,
+      )
+      .toPromise()
```

Comment:
is it possible to refactor this and the get initial messages call into a function?